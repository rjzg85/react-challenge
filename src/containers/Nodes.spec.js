import React from "react";
import { shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { Provider } from "react-redux";
import { create } from "react-test-renderer";
import ConnectedNodes, { Nodes } from "./Nodes";
import Node from "../components/Node";

describe("<Nodes />", () => {
  const actions = {
    checkNodeStatuses: jest.fn(),
  };
  const actionsBlock = {
    getBlocks: jest.fn(),
  };

  const nodes = {
    list: [
      {
        url: "https://thawing-springs-53971.herokuapp.com",
        online: false,
        name: "Node 1",
        loading: false,
      },
      {
        url: "https://secret-lowlands-62331.herokuapp.com",
        online: false,
        name: "Node 2",
        loading: false,
      },
    ],
  };
  const blocks = {
    blocks: [
      {
        id: 1,
        types: "blocks",
        attributes: {
          data: "something",
        },
      },
      {
        id: 2,
        types: "blocks 2",
        attributes: {
          data: "something 2",
        },
      },
    ],
    loading: false,
  };

  it("should contain <Node />!!", () => {
    const wrapper = shallow(
      <Nodes actions={actions} nodes={nodes} actionsBlock={actionsBlock} />
    );

    expect(wrapper.find(Node).length).toEqual(2);
  });

  it("should match snapshot", () => {
    const middlewares = [thunk];
    const store = configureMockStore(middlewares)({
      nodes,
      blocks,
    });
    const component = create(
      <Provider store={store}>
        <ConnectedNodes />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it("should match snapshot", () => {
  //   const middlewares = [thunk];
  //   const store = configureMockStore(middlewares)({
  //     nodes,
  //     blockstate,
  //   });
  //   const component = create(
  //     <Provider store={store}>
  //       <ConnectedNodes>
  //         <ConnectedNode blockstate />
  //       </ConnectedNodes>
  //     </Provider>
  //   );
  //   const tree = component.toJSON();

  //   expect(tree).toMatchSnapshot();
  // });
});
