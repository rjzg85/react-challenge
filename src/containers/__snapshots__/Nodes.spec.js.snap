// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nodes /> should match snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-2"
>
  <h1
    className="MuiTypography-root MuiTypography-h4"
  >
    <strong
      style={
        Object {
          "color": "#000",
        }
      }
    >
      Nodes
    </strong>
  </h1>
  <div
    className="MuiPaper-root MuiExpansionPanel-root makeStyles-root-33 MuiExpansionPanel-rounded MuiPaper-elevation3 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root makeStyles-summary-34"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content makeStyles-content-37"
      >
        <div
          className="MuiBox-root MuiBox-root-83 makeStyles-summaryContent-35"
        >
          <div
            className="MuiBox-root MuiBox-root-84"
          >
            <h5
              className="MuiTypography-root makeStyles-heading-39 MuiTypography-h5"
            >
              Node 1
            </h5>
            <h6
              className="MuiTypography-root makeStyles-secondaryHeading-40 MuiTypography-subtitle1"
            >
              https://thawing-springs-53971.herokuapp.com
            </h6>
          </div>
          <div
            className="MuiBox-root MuiBox-root-89"
          >
            <span
              className="makeStyles-dot-85 makeStyles-dot-87"
            />
            <span
              className="makeStyles-text-86 makeStyles-text-88"
            >
              OFFLINE
            </span>
          </div>
        </div>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon makeStyles-icon-36 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onClick={undefined}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            color={undefined}
            focusable="false"
            role={undefined}
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby={undefined}
            id={undefined}
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root"
            >
              <div
                className="makeStyles-blocksContainer-41"
              >
                <div
                  className="MuiGrid-root makeStyles-blockContainer-114 MuiGrid-container"
                >
                  <div
                    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12"
                  >
                    <p
                      className="MuiTypography-root makeStyles-title-115 MuiTypography-body1"
                    >
                      001
                    </p>
                    <p
                      className="MuiTypography-root makeStyles-body-116 MuiTypography-body1"
                    >
                      something
                    </p>
                  </div>
                </div>
                <div
                  className="MuiGrid-root makeStyles-blockContainer-114 MuiGrid-container"
                >
                  <div
                    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12"
                  >
                    <p
                      className="MuiTypography-root makeStyles-title-115 MuiTypography-body1"
                    >
                      002
                    </p>
                    <p
                      className="MuiTypography-root makeStyles-body-116 MuiTypography-body1"
                    >
                      something 2
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiExpansionPanel-root makeStyles-root-33 MuiExpansionPanel-rounded MuiPaper-elevation3 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root makeStyles-summary-34"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content makeStyles-content-37"
      >
        <div
          className="MuiBox-root MuiBox-root-220 makeStyles-summaryContent-35"
        >
          <div
            className="MuiBox-root MuiBox-root-221"
          >
            <h5
              className="MuiTypography-root makeStyles-heading-39 MuiTypography-h5"
            >
              Node 2
            </h5>
            <h6
              className="MuiTypography-root makeStyles-secondaryHeading-40 MuiTypography-subtitle1"
            >
              https://secret-lowlands-62331.herokuapp.com
            </h6>
          </div>
          <div
            className="MuiBox-root MuiBox-root-224"
          >
            <span
              className="makeStyles-dot-85 makeStyles-dot-222"
            />
            <span
              className="makeStyles-text-86 makeStyles-text-223"
            >
              OFFLINE
            </span>
          </div>
        </div>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon makeStyles-icon-36 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onClick={undefined}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            color={undefined}
            focusable="false"
            role={undefined}
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby={undefined}
            id={undefined}
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root"
            >
              <div
                className="makeStyles-blocksContainer-41"
              >
                <div
                  className="MuiGrid-root makeStyles-blockContainer-114 MuiGrid-container"
                >
                  <div
                    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12"
                  >
                    <p
                      className="MuiTypography-root makeStyles-title-115 MuiTypography-body1"
                    >
                      001
                    </p>
                    <p
                      className="MuiTypography-root makeStyles-body-116 MuiTypography-body1"
                    >
                      something
                    </p>
                  </div>
                </div>
                <div
                  className="MuiGrid-root makeStyles-blockContainer-114 MuiGrid-container"
                >
                  <div
                    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12"
                  >
                    <p
                      className="MuiTypography-root makeStyles-title-115 MuiTypography-body1"
                    >
                      002
                    </p>
                    <p
                      className="MuiTypography-root makeStyles-body-116 MuiTypography-body1"
                    >
                      something 2
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
